### PR TITLE
Moving from hpack to cabal-fmt

### PR DESCRIPTION
**Pull request by kpadmasola**

### CODE REVIEW

File: lib/Eclair/Parser.hs [hunk 1]
```diff
@@ -78,7 +78,7 @@ runParserT path text p = do
   pure $ case result of
     P.Error fatalError ->
       (Nothing, Just $ ParsingError $ toBundle $ fatalError :| P.stateParseErrors s')
-    P.OK x ->
+    P.OK _ x ->
```
[the remark this answers is not in the document]
Comment (luc-tielen) [reply lib/Eclair/Parser.hs:81]: Weird, megaparsec must've chosen a newer version that has a breaking change somehow.

**Review by luc-tielen — APPROVED** [review]
A few small comments, but looks good! Thank you.

File: docs/getting_started.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ versions.
 
 #### Installing the Haskell toolchain
 
-Run the following commands to install `ghcup`, `ghc` and `cabal`. `hpack` and
```
[the remark this answers is not in the document]
Comment (kpadmasola) [reply docs/getting_started.md:18]: done

File: docs/getting_started.md [hunk 3]
```diff
@@ -24,7 +24,6 @@ $ curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
 $ ghcup tui
 # In the terminal UI, select GHC 9.2.4, Haskell language server 1.8 and Cabal 3.6.
 # Important: both install + set them!
-$ cabal install hpack
```
[the remark this answers is not in the document]
Comment (kpadmasola) [reply docs/getting_started.md:27]: done

File: docs/getting_started.md [hunk 4]
```diff
@@ -103,10 +102,9 @@ $ ghcup tui
 ```
 
 The following commands are only needed when working on the compiler.
-Run the commands to install `hpack`, `hspec-discover`, and `hlint`.
```
[the remark this answers is not in the document]
Comment (kpadmasola) [reply docs/getting_started.md:106]: done

File: lib/Eclair/Parser.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (luc-tielen) [reply lib/Eclair/Parser.hs:81]: No it should be fine. But maybe we need to make megaparsec dependency stricter in the future so cabal won't potentially download a incompatible version. 

Not that urgent though, leave it as-is.

**Pull request merged**